es6-lib/errors/index.js

```diff
@@ -0,0 +1,110 @@
+import _ from 'underscore';
+import changeCase from 'change-case';
+
+class GeoError extends Error {
+
+  constructor(parameters) {
+    super();
+    this._params = parameters || {};
+    if (!_.every(this.required, (key) => !!this._params[key])) {
+      throw new Error(`Missing required parameter for ${this.t} error!`);
+    }
+  }
+
+  toString() {
+    return JSON.stringify(this.toJSON());
+  }
+
+  get t() {
+    return changeCase.snakeCase(this.constructor.name);
+  }
+
+  toJSON() {
+    return _.extend({
+      'type': this.t,
+    }, this.parameters);
+  }
+
+  get parameters() {
+    return this._params;
+  }
+
+  get statusCode() {
+    return 400;
+  }
+}
+
+
+class UninterpretableValue extends GeoError {
+  get required() {
+    return ['record', 'filename'];
+  }
+}
+
+class CorruptShapefile extends GeoError {
+  get required() {
+    return ['filename'];
+  }
+}
+
+class CorruptArchive extends GeoError {
+  get required() {
+    return ['reason'];
+  }
+}
+
+class IncompleteArchive extends GeoError {
+  get required() {
+    return ['missing'];
+  }
+}
+
+class ParseError extends GeoError {
+  get required() {
+    return ['lineNumber', 'reason'];
+  }
+}
+
+class RecordTooLarge extends GeoError {
+  get required() {
+    return ['limit', 'nearRecord'];
+  }
+}
+
+class UnknownFileType extends GeoError {
+  get required() {
+    return ['extension'];
```

Comment:
yea you're right about this. I was trying to shoehorn the contentType error into the "UnknownFileType" error class, which is a common error with Di2. The end result is just confusing. The right way to do this would be to add an explicit error and then add the error to the localization file.